### PR TITLE
Introduction to javascript Modules

### LINKED ISSUE #1396: Getting Started With Javascript Modules
### Brief Summary: 
As an application grows bigger, we want to split it into multiple files, so called “modules”. 
A module may contain a class or a library of functions for a specific purpose.

### Key Takeaways: 
1.Core Module features.
2.Module patterns.
3. How to use CommonJS and AMD to manage modules.
4. Bundling Modules
### References: 
N/A

### PR DESCRIPTION
**Pull request by jumamiller**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
This fixes #1396

### CODE REVIEW

**Review by botahamec — CHANGES_REQUESTED** [review]
Hello @jumamiller. I'm requesting some changes to this.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by jumamiller** [comment]
@botahamec  changes requested done!

File: articles/introduction-to-javascript-modules/index.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Now that we have exported several items from the previous part,```Export Module Feature```, in this section, we want to see how to use these exported items in our application.  
+
+Previously, we created a file ```add.js```, let's see how we can import the contents of this module into our new module file and use its contents.  
+
+Let's start by creating a file called ```results.js```.  
+
+```js
+//import the contents of add.js
+//ensure the file path is correct
+import { addTwoNumbers } from './modules/add.js';
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md:75]: I still don't see anything indicating this. Correct me if I'm wrong

File: articles/introduction-to-javascript-modules/index.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+```
+Well, this is slightly different when working with modules, you need a few modifications as we will see soon.  
+
+The first and the most important part is to add a ``` type``` as ```module```, that is, ```type=" module" ``` in the script element.  
+This ensures that the script is defined or declared as a module.  
+
+Now to import ```results.js``` we would do something like:-  
+```html
+<head>
+    <script type="module" src="results.js"></script>
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md:109]: I don't think this was fixed. Again, correct me if I'm wrong though

File: articles/introduction-to-javascript-modules/index.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+</head>
+<body>
+    <button onClick="addTwoNumbers(1,2);">
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md:112]: This doesn't seem to have been fixed either.

File: articles/introduction-to-javascript-modules/index.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+     +", admission number "+ admission_number
+     +". I'm in year"+ yearOfStudy;
+
+    return myDetails;
+}
+
+export { studentName,studentDetails };
+
+```
+In the above code, we have a method named ```studentDetails()```  which is being passed several parameters and, a variable named ```studentName```.
```
Comment (botahamec) [inline articles/introduction-to-javascript-modules/index.md:58]: The comma goes before the word "and".

Mentioning the parameters of the function seems irrelevant.

```suggestion
In the above code, we have a method named ```studentDetails()```, and a variable named ```studentName```.

**Comment by jumamiller** [comment]
lemme push, will ping you,seems i didn't commit changes

File: articles/introduction-to-javascript-modules/index.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+function studentDetails(name,
+                        admission_number,
+                        college,yearOfStudy
+)
+{
+    const myDetails="My name is "+ name
+     + " ,a student at "+ college
+     + " ,admission number "+ admission_number
+     +" .I'm in year"+ yearOfStudy;
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md]: I'm unresolving this comment to remind you that there should be a space after the word, "year"

File: articles/introduction-to-javascript-modules/index.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    <script type="module" src="results.js"></script>
+</head>
+<body>
+    <button onClick="addTwoNumbers(1,2);">
+    Calculate Sum</button>
+</body>
+```
+
+
+NOTE:- The Javascript module could also be embedded directly into the HTML file as follows:-
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md]: One dash has been removed but the other dash is still there.

File: articles/introduction-to-javascript-modules/index.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+In our ```example.html``` file, add the following.  
+```html
+<head>
+    <script type="text/javascript" src="myScript.js"></script>
+</head>
+
+```
+Well, this is slightly different when working with modules, you need a few modifications as we will see soon.  
+
+The first and the most important part is to add a ``` type``` as ```module```, that is, ```type=" module" ``` in the script element.
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md]: The spacing has changed, but the sentence itself hasn't

File: articles/introduction-to-javascript-modules/index.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+```js
+//import the contents of add.js
+//ensure the file path is correct
+import { addTwoNumbers } from './modules/add.js';
+
+```
+
+In the file above, we only have 1 line of code, the ```import``` statement.  
+What happens is that the import statement lists the items which are being imported and including the path where the module is located.  
+Once these items are imported into the file, in this case, ```results.js```, they are readily available for using a smart IDE will even suggest for you the methods and properties when you start typing.
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md]: I don't understand the change you made at all. You replaced "using" with "use", but you also replace "a" with "of"? And it's still a run-on sentence.

File: articles/introduction-to-javascript-modules/index.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+function studentDetails(name,
+                        admission_number,
+                        college,yearOfStudy
+)
+{
+    const myDetails="My name is "+ name
+     + " ,a student at "+ college
+     + " ,admission number "+ admission_number
+     +" .I'm in year"+ yearOfStudy;
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/introduction-to-javascript-modules/index.md]: The space before the period is fixed, but a space after "year" is still needed.

**Review by botahamec — CHANGES_REQUESTED** [review]
@jumamiller A lot of the comments you've marked as resolved are not actually resolved. Please fix those.

In addition, I have a new comment to a change you made, and also should help with shortening the sentence down.

**Comment by jumamiller** [comment]
@botahamec kindly lemme push changes, I'll let you know. Thanks

**Comment by botahamec** [comment]
Oh I see your other comment now

**Comment by jumamiller** [comment]
@botahamec, all the 'changes' you requested have been made. I have personally made sure every single line of my article is edited on both Grammarly and Hemingway Editor.
Mr. @botahamec, each script has been run on both Firefox and chrome, they are working. Kindly take your time, run each script on your machine following the tutorial. In case of any issue, rather than having all these chats on the PR, ping me on the slack.
Thanks.

**Review by botahamec — APPROVED** [review]
@jumamiller You did a great job of editing this article.

@hectorkambow This is ready

**Pull request merged**